gocourse03/hw/main.go

```diff
@@ -0,0 +1,241 @@
+package main
+
+import (
+	"fmt"
+	"os"
+)
+
+type Human struct {
+	Name string
+	NameMemory bool
+	GlobalMemory bool
+	Bag bool
+}
+
+type Cave struct {
+	Entry string
+	Dark bool
+}
+
+type Backpack struct {
+	Size int
+	Matches *Matches
+	Flashlight *Flashlight
+	Knife *Knife
+}
+
+type Matches struct {
+	Item string
+	Exist bool
+}
+
+type Flashlight struct {
+	Item string
+	Exist bool
+}
+
+type Knife struct {
+	Item string
+	Exist bool
+}
+
+type Road struct {
+	Destination string
+}
+
+type Body struct {
+	IsDead bool
+	Type string
+}
+
+type Camp struct {
+	HasPeople bool
+}
+
+type Bug struct {
+	IsBig bool
+}
+
+func (b Bug) Bite() {
+	if b.IsBig {
+		fmt.Printf("Big bug bites her victim and it passes out!")
+	} else {
+		fmt.Printf("Small bug bites her victim and it just hearts!")
+	}
+} 
+
+func (h *Human) FindBag(backpack Backpack) {
+	switch h.Bag {
+	case true:
+		fmt.Printf("%s finds bag.\n", h.Name)
+		fmt.Printf("%s's bag contains:\n", h.Name)
+		if backpack.Matches.Exist == true {
+			fmt.Printf("- Matches\n") 
+		} else if backpack.Matches.Exist == false {
+				fmt.Printf("- No Matches\n") 
+		}
+		if backpack.Flashlight.Exist == true {
+			fmt.Printf("- Flashlight\n") 
+		} else if backpack.Flashlight.Exist == false {
+			fmt.Printf("- No Flashlight\n") 
+		}
+		if backpack.Knife.Exist == true {
+			fmt.Printf("- Knife\n") 
+		} else if backpack.Knife.Exist == false {
+			fmt.Printf("- No Knife\n") 
+		}
+	case false:
+		fmt.Printf("%s bag is lost.\n", h.Name)
+	}
+}
+
+func (h Human) Choice(choice string) {
+	fmt.Printf(choice)
+}
+
+var a int
+
+func main() {
+	men := &Human{
+		Name: "Steven", 
+		NameMemory: false,
+		Bag: false,
+	}
+
+	cave := Cave{"big cage entrance", true}
+
+	bag := Backpack{
+		Size: 10,
+		Matches: &Matches{"Matches", false},
+		Flashlight: &Flashlight{"Flashlight", false},
+		Knife: &Knife{"Knife", false},
+	}
+
+	fmt.Printf("%s wokes up near %s.\n", men.Name, cave.Entry)
+
+	fmt.Printf("Does %s remember his name? 1 or 2?\n", men.Name)
+	fmt.Scan(&a)
+	switch a {
+	case 1:
+		men.NameMemory = true
+		fmt.Printf("%s remembers his name.\n", men.Name)
+	case 2:
+		men.NameMemory = false
+		fmt.Printf("%s doesn't remember his name.\n", men.Name)
+	}
+	fmt.Println()
+
+	fmt.Printf("Does %s find a bag? 1 or 2?\n", men.Name)
```

Comment:
I think we should be displaying short description for each choice instead of forcing the player to guess